app/src/main/java/com/github/se/gomeet/ui/mainscreens/EventInfo.kt

```diff
@@ -0,0 +1,308 @@
+          contentDescription = "Email",
+          modifier = Modifier.size(30.dp),
+          tint = DarkCyan)
+    }
+  }
+}
+
+@Composable
+fun MapViewComposable(
+    loc: LatLng,
+    zoomLevel: Float = 15f // Default zoom level for close-up of location
+) {
+  val cameraPositionState = rememberCameraPositionState {
+    position = CameraPosition.fromLatLngZoom(loc, zoomLevel)
+  }
+
+  val markerState = rememberMarkerState(position = loc)
+
+  // Set up the GoogleMap composable
+  GoogleMap(
+      modifier = Modifier.fillMaxWidth().height(200.dp).clip(RoundedCornerShape(20.dp)),
+      cameraPositionState = cameraPositionState) {
+        Marker(
+            state = markerState,
+            title = "Marker in Location",
+            snippet = "This is the selected location")
+      }
+
+  // Initialize the map position once and avoid resetting on recomposition
+  DisposableEffect(loc) {
+    cameraPositionState.move(CameraUpdateFactory.newLatLngZoom(loc, zoomLevel))
+    onDispose {}
+  }
+
+  DisposableEffect(loc) {
+    markerState.position = loc
+    onDispose {}
+  }
+}
+
+@Composable
+fun EventInfoScreen(navController: NavHostController) {
+  val backStackEntry = navController.currentBackStackEntryAsState().value
+  val arguments = backStackEntry?.arguments
+
+  val title = arguments?.getString("title") ?: ""
+  val date = arguments?.getString("date") ?: ""
+  val time = arguments?.getString("time") ?: ""
+  val organizer = arguments?.getString("organizer") ?: ""
+  val rating = arguments?.getDouble("rating") ?: 0.0
+  val description = arguments?.getString("description") ?: ""
+  val latitude = arguments?.getFloat("latitude") ?: 0.0
+  val longitude = arguments?.getFloat("longitude") ?: 0.0
+  val loc = LatLng(latitude.toDouble(), longitude.toDouble())
+
+  Log.d("EventInfoScreen", "Loc is $loc")
+
+  EventInfo(
+      nav = NavigationActions(navController),
+      title = title,
+      date = date,
+      time = time,
+      organizer = organizer,
+      rating = rating,
+      image =
+          painterResource(
+              id = R.drawable.chess_demo), // Image handling might need different approach
+      description = description,
+      loc = loc)
+}
+
+@Preview(showBackground = true)
+@Composable
+fun EventScreenPreview() {
+  EventInfo(
+      nav = NavigationActions(rememberNavController()),
+      title = "Chess Tournament",
+      organizer = "EPFL Chess Club",
+      date = "TUE",
+      time = "19:10",
+      rating = 4.8,
+      image = painterResource(id = R.drawable.chess_demo),
+      description =
+          "Howdy!\n\nAfter months of planning, La Dame Blanche is finally offering you a rapid tournament!\n\nJoin us on Saturday 23rd of March afternoon for 6 rounds of 12+3” games in the chill and cozy vibe of Satellite. Take your chance to have fun and play, and maybe win one of our many prizes\n\nOnly 50 spots available, with free entry!",
+      loc = LatLng(46.519962, 6.633597))
+}
```

Comment:
Overall the EventInfo screen looks fine to me.
The only issues I see are hardcoded values for padding. Sometimes these are small values so it should be fine. However, I think spacers can lead to issues when testing on smaller screens.
Also shouldn't we use Roboto for all texts in order to be consistent with the rest of the app?

Reply:
Page is scrollable to no problem for spacers and roboto isn't working well on titles